### DISCUSSION TITLE
Saving parallell files

### DISCUSSION THREAD

**Discussion by rnpratoori** [pyvista/pyvista#3719] [Q&A]
Hello team,

I have a bunch of *pvtu files and each of them has ~200 *vtu files linked. I need to do some post-processing where I will be adding and deleting point data for variables. As of now, I do it by reading the files through pyvista and doing the post-processing, and saving them as single *vtu files. I want to save them in *pvtu format so that I can take advantage of the parallel processing. What is the most efficient was to go about it?

**Comment by MatthewFlamm** [comment]
I think having more full featured writing classes and helpers is a good feature request.  If you are familiar with vtk, you can use this class to do what you want:

https://vtk.org/doc/nightly/html/classvtkXMLPUnstructuredGridWriter.html

Here is some code that seems to work as a pointer, but I don't think it takes advantage of the parallel processing yet, but it is a start:

```py
import vtk
from pyvista import examples
a = examples.download_cylinder_crossflow()
writer = vtk.vtkXMLPUnstructuredGridWriter()
writer.SetInputData(a[0])
writer.SetNumberOfPieces(2)
writer.SetFileName("test.pvtu")
writer.SetStartPiece(0)
writer.SetEndPiece(1)
writer.Write()
```

**Reply by rnpratoori** [reply]
@MatthewFlamm Thank you for your reply. This is saving the files in *pvtu, but the size of the *vtu files is as big as if they are saved directly as a single *vtu file. When I initially wrote *pvtu files from deal.ii, the size of *vtu files is much smaller with at least 3 times larger data set. I am attaching the 2 *pvtu files for reference. solution* is the one generated originally from deal.ii and epst_solution* is the one I generated using the above code. 
[pvtu.zip](https://github.com/pyvista/pyvista/files/10300051/pvtu.zip)

**Reply by MatthewFlamm** [reply]
You only zipped the pvtu file, and it is missing the actual data. The pvtu files themselves just have pointers to the data files and you have different number of data files. 

I'm wondering if there is a difference in compression or ASCII vs binary.
